src/Raven.Server/RavenServer.cs

```diff
@@ -929,45 +938,179 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.ClusterNode)
-                        {
-                            authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.Operator)
+                        var cert = ServerStore.Cluster.GetCertificateByThumbprint(ctx, certificate.Thumbprint) ??
+                                   ServerStore.Cluster.GetLocalStateByThumbprint(ctx, certificate.Thumbprint);
+
+                        if (cert == null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Operator;
+                            // If connection will be approved, cert won't be null anymore and will be assigned the relevant permissions
+                            MaybeAllowConnectionBasedOnPinningHash(certificate, ctx, ref authenticationStatus, ref cert, remoteAddress);
                         }
-                        else
+
+                        if (cert != null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Allowed;
-                            foreach (var kvp in definition.Permissions)
+                            var definition = JsonDeserializationServer.CertificateDefinition(cert);
+                            authenticationStatus.Definition = definition;
+                            if (definition.SecurityClearance == SecurityClearance.ClusterAdmin)
                             {
-                                authenticationStatus.AuthorizedDatabases.Add(kvp.Key, kvp.Value);
+                                authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
+                            }
+                            else if (definition.SecurityClearance == SecurityClearance.ClusterNode)
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
+                            }
+                            else if (definition.SecurityClearance == SecurityClearance.Operator)
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.Operator;
+                            }
+                            else
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.Allowed;
+                                foreach (var kvp in definition.Permissions)
+                                {
+                                    authenticationStatus.AuthorizedDatabases.Add(kvp.Key, kvp.Value);
+                                }
                             }
                         }
                     }
+                    finally
+                    {
+                        ctx.CloseTransaction();
```

Comment:
why do we need to explicitly close the transaction here? Isn't this closed by the `using` above?

Reply:
Fixed